### DISCUSSION TITLE
Accessing request & response in the handler

### DISCUSSION THREAD

**Discussion by JUNIORCO** [RobinTail/express-zod-api#925] [Q&A] [answered]
Use case: one of my endpoints needs to access cookies (`req.get(x)`), set cookies (`res.cookie(x, y)`), and sometimes redirect (`res.redirect(url)`).

What's the cleanest way to do this?

**Comment by RobinTail** [comment]
Hello @JUNIORCO ,

Take a look on the first example of Middlewares in the documentation:

https://github.com/RobinTail/express-zod-api#middlewares

Thus, you can make your endpoint handler aware of certain properties of `request` via `options`.

For redirecting you should make a ResultHandler, which has direct access to `request` and `response`.

By the way, there was another discussion on Cookies. Check this out:
https://github.com/RobinTail/express-zod-api/discussions/710#discussioncomment-4257890

**Comment by RobinTail** [comment, marked as answer]
First-class cookie support added in v28.1.0, @JUNIORCO 
https://ez.robintail.cz/v28.1.0/cookies

**Reply by anthonyharper834-web** [reply]
How can I chat with No Ko doll #3208?
